src/ddsrt/CMakeLists.txt

```diff
@@ -270,8 +270,7 @@ if(${CMAKE_C_COMPILER_ID} STREQUAL "SunPro")
   target_link_libraries(ddsrt INTERFACE socket nsl)
 endif()
 
-if(ENABLE_SHM)
-  find_package(iceoryx_binding_c REQUIRED)
+if(iceoryx_binding_c_FOUND)
   target_link_libraries(ddsrt INTERFACE iceoryx_binding_c::iceoryx_binding_c)
 endif()
```

Comment:
On behalf of @dkroenke.
For the time being it is required since some functionality to enable shared memory is currently there and requires iceoryx. This will be refactored as soon as possible. I talked about the issue with @eboasson , we will be able move all dependencies to ddsi or ddsc later. So this is a somewhat intermediate stage.